test/gtest/ucp/test_ucp_tag.cc

```diff
@@ -423,22 +424,133 @@ class test_ucp_tag_limits : public test_ucp_tag {
         test_ucp_tag::init();
+
         check_offload_support(m_test_offload);
+
+        if (m_test_offload) {
+            ucp_ep_config_t *cfg = ucp_ep_config(sender().ep());
+            m_min_rndv = ucp_ep_tag_offload_min_rndv_thresh(sender().ucph(),
+                                                            &cfg->key);
+        }
     }
 
     static void get_test_variants(std::vector<ucp_test_variant>& variants) {
-        add_variant_with_value(variants, get_ctx_params(), 0, "");
-        add_variant_with_value(variants, get_ctx_params(), 1, "offload");
+        ucp_params_t params = get_ctx_params();
+        params.features = UCP_FEATURE_TAG | UCP_FEATURE_AM;
+
+        add_variant_with_value(variants, params, 0, "");
+        add_variant_with_value(variants, params, 1, "offload");
     }
 
 protected:
-    bool m_test_offload;
+    bool    m_test_offload;
+    size_t  m_min_rndv;
+
+    static void check_short_thresh(const ucp_memtype_thresh_t &thresh,
+                                   size_t cfg_thresh)
+    {
+        EXPECT_LE(thresh.memtype_on + 1, cfg_thresh);
+        EXPECT_LE(thresh.memtype_off + 1, cfg_thresh);
+    }
+
+    void check_rndv_startup_config(size_t exp_rndv_thresh,
+                                   size_t exp_rndv_intra_thresh,
+                                   size_t exp_rndv_inter_thresh)
+    {
+        ucp_context_config_t *cfg = &sender().worker()->context->config.ext;
+
+        EXPECT_EQ(exp_rndv_thresh, cfg->rndv_thresh);
+        EXPECT_EQ(exp_rndv_intra_thresh, cfg->rndv_intra_thresh);
+        EXPECT_EQ(exp_rndv_inter_thresh, cfg->rndv_inter_thresh);
+    }
+
+    void check_tag_rndv_v2(size_t cfg_thresh)
+    {
+        ucp_ep_config_t *cfg = ucp_ep_config(sender().ep());
+
+        if (m_test_offload) {
+            /* If configured threshold is less than min_rndv, then expect exact
+             * min_rndv limit for short messages */
+            if (cfg_thresh < m_min_rndv) {
+                EXPECT_EQ(cfg->tag.offload.max_eager_short.memtype_on + 1,
+                          m_min_rndv);
+                EXPECT_EQ(cfg->tag.offload.max_eager_short.memtype_off + 1,
+                          m_min_rndv);
+            } else {
+                check_short_thresh(cfg->tag.offload.max_eager_short, cfg_thresh);
+            }
+        } else {
+            check_short_thresh(cfg->tag.max_eager_short, cfg_thresh);
+        }
+    }
+
+    void check_am_rndv_v2(size_t cfg_thresh)
+    {
+        ucp_ep_config_t *cfg = ucp_ep_config(sender().ep());
+
+        check_short_thresh(cfg->am_u.max_eager_short, cfg_thresh);
+        check_short_thresh(cfg->am_u.max_reply_eager_short, cfg_thresh);
+    }
+
+    void check_ep_proto_rndv(size_t cfg_thresh, bool expect_rndv)
+    {
+        ucp_ep_config_t *cfg = ucp_ep_config(sender().ep());
+        ucp_proto_select_t *proto_select = &cfg->proto_select;
+        ucp_proto_select_elem_t value;
+        size_t msg_length = cfg_thresh;
```

Comment:
According to coding style initialized vars go first:
```suggestion
        ucp_ep_config_t *cfg = ucp_ep_config(sender().ep());
        ucp_proto_select_t *proto_select = &cfg->proto_select;
        size_t msg_length = cfg_thresh;
        ucp_proto_select_elem_t value;
```